cmd/cfssl/cfssl_test.go

```diff
@@ -1 +1,446 @@
+				OU: "Certificate Authority",
+			},
+		},
+		Hosts:      []string{"ca.example.com"},
+		KeyRequest: &keyRequest,
+		CA:         &CAConfig,
+	}
+)
+
+// We will Unmarshal API and CLI outputs into the following structs
+
+// Bundle resulting struct is not defined in bundler/bundle.go. A custom
+// bundle struct will be made here for convenience.
+type CLIBundleResponse struct {
+	Bundle string `json:"bundle"`
+	CRT    string `json:"crt"`
+	Root   string `json:"root"`
+}
+
+type APIBundleResponse struct {
+	Success bool `json:"success"`
+	Result  struct {
+		CLIBundleResponse
+	} `json:"result"`
+}
+
+// Scan struct utilizes ScannerResult struct defined in
+// scan/scan_common.go.
+type CLIScanResponse struct {
+	Connectivity struct {
+		CloudFlareStatus struct {
+			scan.ScannerResult
+		} `json:"CloudFlareStatus"`
+		DNSLookup struct {
+			scan.ScannerResult
+		} `json:"DNSLookup"`
+	} `json:"Connectivity"`
+	PKI struct {
+		ChainExpiration struct {
+			scan.ScannerResult
+		} `json:"ChainExpiration"`
+		ChainValidation struct {
+			scan.ScannerResult
+		} `json:"ChainValidation"`
+		CipherSuite struct {
+			scan.ScannerResult
+		} `json:"CipherSuite"`
+	} `json:"PKI"`
+	TLSHandshake struct {
+		CertsByCiphers struct {
+			scan.ScannerResult
+		} `json:"CertsByCiphers"`
+		CertsBySigAlgs struct {
+			scan.ScannerResult
+		} `json:"CertsBySigAlgs"`
+		SigAlgs struct {
+			scan.ScannerResult
+		} `json:"SigAlgs"`
+	} `json:"TLSHandshake"`
+}
+
+type APIScanResponse struct {
+	Success bool `json:"success"`
+	Result  struct {
+		CLIScanResponse
+	} `json:"result"`
+}
+
+// API Struct for CertInfo utilizes certinfo.Certificate struct.
+type APICertInfoResponse struct {
+	Success bool `json:"success"`
+	Result  struct {
+		certinfo.Certificate
+	} `json:"result"`
+}
+
+// API Struct for Sign and Info will utilize info.Resp struct (for the
+// "certificate" field)
+type APIResponse struct {
+	Success bool `json:"success"`
+	Result  struct {
+		info.Resp
+	} `json:"result"`
+}
+
+// CFSSL Sign on CLI returns a json with a "cert" field, not "certificate"
+// as is with the Sign API response struct defined in scan_common.go.
+// The following struct is made to suit that situation.
+type CLISignResponse struct {
+	Certificate string `json:"cert"`
+}
+
+// Start up the CFSSL local server.
+func setup() {
+	// Set up a test server using our CA certificate and key.
+	CACert, CAKey, err := testsuite.CreateSelfSignedCert(CARequest)
+	if err != nil {
+		panic(err)
+	}
+	// Set up a test server using our CA certificate and key.
+	serverData := testsuite.CFSSLServerData{CA: CACert, CAKey: CAKey}
+	s, err := testsuite.StartCFSSLServer(addressToTest, portToTest,
+		serverData)
+	server = s
+	if err != nil {
+		setdown()
+		panic(err)
+	}
+}
+
+// Kill the CFSSL local server.
+func setdown() {
+	err := server.Kill()
+	if err != nil {
+		panic(err)
+	}
+}
+
+// Return API output
+func apiOut(request string, jsonStr []byte,
+	endpoint string, t *testing.T) []byte {
```

Comment:
I think it would look cleaner to not have the function signature split into two lines